rust/rbac-registration/src/cardano/cip509/mod.rs

```diff
@@ -25,28 +25,42 @@ use validation::{
 use x509_chunks::X509Chunks;
 
 use super::transaction::witness::TxWitness;
-use crate::utils::{
-    decode_helper::{decode_bytes, decode_helper, decode_map_len},
-    general::{decode_utf8, decremented_index},
-    hashing::{blake2b_128, blake2b_256},
+use crate::{
+    cardano::cip509::rbac::Cip509RbacMetadata,
+    utils::{
+        decode_helper::{decode_bytes, decode_helper, decode_map_len},
+        general::decremented_index,
+        hashing::{blake2b_128, blake2b_256},
+    },
 };
 
 /// CIP509 label.
 pub const LABEL: u64 = 509;
 
-/// CIP509.
-#[derive(Debug, PartialEq, Clone, Default)]
+/// A x509 metadata envelope.
+///
+/// The envelope is required to prevent replayability attacks. See [this document] for
+/// more details.
+///
+/// [this document]: https://github.com/input-output-hk/catalyst-CIPs/blob/x509-envelope-metadata/CIP-XXXX/README.md
+#[derive(Debug, PartialEq, Clone)]
 pub struct Cip509 {
-    /// `UUIDv4` Purpose .
-    pub purpose: Uuid, // (bytes .size 16)
+    /// A registration purpose (`UUIDv4`).
+    ///
+    /// The purpose is defined by the consuming dApp.
+    pub purpose: Uuid,
     /// Transaction inputs hash.
-    pub txn_inputs_hash: TxInputHash, // bytes .size 16
-    /// Optional previous transaction ID.
-    pub prv_tx_id: Option<Hash<32>>, // bytes .size 32
-    /// x509 chunks.
-    pub x509_chunks: X509Chunks, // chunk_type => [ + x509_chunk ]
+    pub txn_inputs_hash: TxInputHash,
+    /// An optional `BLAKE2b` hash of the previous transaction.
+    ///
+    /// The hash must always be present except for the first registration transaction.
+    pub prv_tx_id: Option<Hash<32>>,
```

Comment:
I have added a TODO-comment to use a type from the `cardano-blockchain-types` and removed the "BLAKE2b" part from the description entirely because it will be apparent from the type.